packages/core/src/shared/telemetry/spans.ts

```diff
@@ -90,6 +90,40 @@ function getValidatedState(state: Partial<MetricBase>, definition: MetricDefinit
     return missingFields.length !== 0 ? Object.assign({ missingFields }, state) : state
 }
 
+/**
+ * Options used for the creation of a span
+ */
+export type SpanOptions = {
+    /** True if this span should emit its telemetry events. Defaults to true if undefined. */
+    emit?: boolean
+
+    /**
+     * Adds a function entry to the span stack.
+     *
+     * This allows you to eventually retrieve the function entry stack by using {@link TelemetryTracer.getFunctionStack()},
+     * which tells you the chain of function executions to bring you to that point in the code.
+     *
+     * Example:
+     * ```
+     * function a() {
+     *   telemetry.your_Metric.run(() => b(), { functionId: { name: 'a'} })
+     * }
+     *
+     * function b() {
+     *   telemetry.your_Metric.run(() => c(), { functionId: { name: 'b'} })
+     * }
+     *
+     * function c() {
+     *   telemetry.your_Metric.run(() => {
+     *     const stack = telemetry.getFunctionStack()
+     *     console.log(stack) // [ {source: 'a' }, { source: 'b' }, { source: 'c' }]
+     *   }, { functionId: { name: 'c'} })
+     * }
+     * ```
+     */
+    functionId?: FunctionEntry
```

Comment:
I don't quite get why SpanOptions is needed. Why can't its fields live on the `function_call` metric itself?

We have a story to better support "nested" metrics: https://sim.amazon.com/issues/IDE-10178

With that support, each `function_call` could be a child of the containing (parent) `function_call`. And then the "stack" could be derived that way. 

And then we avoid this extra alternative concept.